library/network_connections.py

```diff
@@ -1098,8 +1107,12 @@ def connection_create(self, connections, idx, connection_current=None):
             for nameserver in ip["dns"]:
                 if nameserver["family"] == socket.AF_INET6:
                     s_ip6.add_dns(nameserver["address"])
-            for search_domain in ip["dns_search"]:
-                s_ip6.add_dns_search(search_domain)
+            # NetworkManager disallows to config `ipv6.dns-search` setting when
+            # ipv6.method is disabled or ignore. The role should silently ignore
+            # such setting.
```

Comment:
if the role *always* is supposed to ignore dns-search with IP method disabled, then the right place to perform such normalization is `ArgValidator`. That one is supposed to normalize the user input in a form, that can be used directly (with little additional workarounds) by the provider implementation.

On the other hand, if this is only a special behavior of the `nm` backend, then ArgValidator must not normalize settings away, which might be used by other backends. In that case, the patch is right, but the comment is wrong ("The role should silently ignore such setting" is wrong, it is only the "nm" provider)

Reply:
I looked at the files associated with "DNS search", "ipv6 disabled" in `network-scripts` ( `grep -r 'DOMAIN\|disable_ipv6\|IPV6INIT' $(rpm -ql network-scripts)`), I am convinced that it is only the special behavior of nm backend. I updated the comments, thanks.